### PR TITLE
Enable Mypy Checking in torch/_inductor/freezing.py

### PR DESCRIPTION
**Pull request by Cyril-Anto**
Fixes #105230 

Summary:

As suggested in https://github.com/pytorch/pytorch/issues/105230 mypy checking is enabled in torch/_inductor/freezing.py

After Fix:

mypy --follow-imports=skip torch/_inductor/freezing.py Success: no issues found in 1 source file

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @Xia-Weiwen @ngimel

### CODE REVIEW

**Comment by colesbury** [comment]
@eellison would you please review this?

File: torch/_inductor/freezing.py [hunk 1]
```diff
@@ -133,7 +133,7 @@ def run_node(self, node):
         out = super().run_node(node)
 
         if node.op != "get_attr" and isinstance(out, torch.Tensor):
-            if not self.insertable_tensor_check(out):
+            if not self.insertable_tensor_check(out): # type: ignore[operator]
```
Comment (eellison) [inline torch/_inductor/freezing.py:136]: why do we need this ? shouldn't this be well typed ?

File: torch/_inductor/freezing.py [hunk 2]
```diff
@@ -84,8 +84,8 @@ def __init__(
         insertable_tensor_check: Optional[Callable[[torch.Tensor], bool]] = None,
     ):
         super().__init__(gm)
-        self.node_replacements = {}
-        self.replaced_uses = collections.Counter()
+        self.node_replacements: Dict[int, torch.Tensor] = {}
```
Comment (eellison) [inline torch/_inductor/freezing.py:87]: this should be `[torch.fx.Node, torch.Tensor]`

**Review by eellison — COMMENTED** [review]
looks good, one question

**Comment by Cyril-Anto** [comment]
@eellison Made the changes you suggested, can you please review it once?

File: torch/_inductor/freezing.py [hunk 3]
```diff
@@ -136,8 +136,8 @@ def run_node(self, node):
         out = super().run_node(node)
 
         if node.op != "get_attr" and isinstance(out, torch.Tensor):
-            if not self.insertable_tensor_check(out):
-                return out
+            if not isinstance(out, torch.Tensor):
```
Comment (eellison) [inline torch/_inductor/freezing.py:139]: we're no longer checking :

```
if not self.insertable_tensor_check(out):
```

**Review by eellison — COMMENTED** [review]
has functional changes

**Pull request closed without merging**